frontend/src/pages/user/result.js

```diff
@@ -1,39 +1,65 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
+import { useParams } from "react-router-dom";
+
 import ResultEntry from "../../components/resultEntry";
+import { getResult } from "../../services/lessonService";
 
 const Result = () => {
-  return (
+  const { id } = useParams();
+  const [state, setState] = useState();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getResult(id)
+      .then(({ data }) => {
+        setState(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        setLoading(false);
+      });
```

Comment:
```suggestion
    getResult(id)
      .then(({ data }) => {
        setState(data);
      })
      .finally(() => setLoading(false));
```
Since setLoading can be both, you can use finally and call it once